### ISSUE TITLE
[vminsert]:  `checkReadOnlyMode` method always sets `isReadOnly` to 0

### ISSUE THREAD

**Issue by z-anshun** [VictoriaMetrics/VictoriaMetrics#4870]
### Is your question request related to a specific component?

vminsert

### Describe the question in detail

Hi, team,
When I read the source code of netstorage, I found a strange logic.

https://github.com/VictoriaMetrics/VictoriaMetrics/blob/cluster/app/vminsert/netstorage/netstorage.go#L806-L811
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/cluster/app/vminsert/netstorage/netstorage.go#L322-L326

According to the above code. `checkReadOnlyMode` is called every 30s, but `sendToConn` called in `checkReadOnlyMode` always returns `nil`, so `sn.isReadOnly` will be set to 0 every 30s.

### Troubleshooting docs

- [ ] General - https://docs.victoriametrics.com/Troubleshooting.html
- [ ] vmagent - https://docs.victoriametrics.com/vmagent.html#troubleshooting
- [ ] vmalert - https://docs.victoriametrics.com/vmalert.html#troubleshooting

**Comment by f41gh7** [comment]
Hello, it's a bug. For now `checkReadOnlyMode` performs inefficient read only check.

I think, at first new health check API must be introduced for vmstorage. vminsert must call it for actual read only check.

**Comment by z-anshun** [comment]
@valyala , Okay, I did find some problems.
During the verification process, I first processed some `vmstorage` nodes as read-only. At this time, `vminsert` will not write to them. However, when I try to disconnect a `vmstorage`, even if the `vmstorage` node returns to normal, No data will be written either.
Then I looked at the code and found that although the `broken` field of `storageNode` was 0, the `isReadOnly` field was always set to 1, causing the `isReady` method to always return false. And the checkHealth method will restore the connection between `vminsert` and `vmstorage` only when `bc` is nil.
The relevant code is as follows:
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/71aee14ff03afd0b3e2855e49c47174409b676a6/app/vminsert/netstorage/netstorage.go#L254-L262
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/71aee14ff03afd0b3e2855e49c47174409b676a6/app/vminsert/netstorage/netstorage.go#L294-L317
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/71aee14ff03afd0b3e2855e49c47174409b676a6/app/vminsert/netstorage/netstorage.go#L810-L818

Therefore, I think the following logic needs to be added, or other better solutions
```go
if sn.bc != nil && atomic.LoadUint32(&sn.isReadOnly) == 0 {
		// The sn looks healthy.
		return
}
bc, err := sn.dial()
```

**Comment by f41gh7** [comment]
I think, an issue related to the incorrect usage of `sendToConn` at the `checkReadOnlyMode` function. It must set bc to nil in case of error. I'll send PR with fix.

**Comment by f41gh7** [comment]
@z-anshun can you try to build vminsert from commit https://github.com/VictoriaMetrics/VictoriaMetrics/pull/4935/commits/2329b5538619b41a92fdec32b570a678082526bb and check if it fixes your issue?

**Comment by z-anshun** [comment]
> @z-anshun can you try to build vminsert from commit [2329b55](https://github.com/VictoriaMetrics/VictoriaMetrics/commit/2329b5538619b41a92fdec32b570a678082526bb) and check if it fixes your issue?

@f41gh7 Okay, I tested and everything is OK .

**Comment by valyala** [comment]
FYI, the issue has been fixed in [v1.87.8 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.87.8). It is also going to be fixed in the upcoming v1.93.3 LTS release.

**Comment by valyala** [comment]
The issue has been fixed in [v1.93.3 release](https://docs.victoriametrics.com/CHANGELOG.html). Closing it then.

@z-anshun , thanks for the help!
